src/server/api/assignment.js

```diff
@@ -667,13 +692,18 @@ export async function giveUserMoreTexts(
           return countUpdated;
         }
       }
+
+      const { teamId, teamTitle } = team;
+      teamsAssignedTo.set(teamId, teamTitle);
     }
 
     return countUpdated;
   });
 
   // Async function, not awaiting because response to external assignment tool does not depend on it
-  notifyIfAllAssigned(assignmentInfo.type, auth0Id, organizationId);
+  notifyIfAllAssigned(organizationId, teamsAssignedTo, parentTrx).catch(err =>
```

Comment:
I don't think that this should happen within the same transaction

• It's dangerous to extend a transaction without an `await`. In this case, since this function could return before `parentTrx` is used, parentTrx may be called after the transaction has committed. I don't know what would happen under that circumstance!
• If `parentTrx` is rolled back, we shouldn't notify

To ensure that the transaction has committed, we could either:
A) Queue this outside of `giveUserMoreTexts`, passing required info from `giveUserMoreTexts` back to its caller
B) Throw this in a setTimeout with like 15 seconds

B is easier and I think fine – A has the benefit of opting not to run (saving a database round trip, client usage, etc.) if nothing was assigned